### DISCUSSION TITLE
How to catch started rollout event?

### DISCUSSION THREAD

**Discussion by danil-smirnov** [argoproj/argo-rollouts#3210] [Q&A] [answered]
As per docs, we have some notifications triggers [predefined](https://argo-rollouts.readthedocs.io/en/release-1.5/features/notifications/#default-trigger-templates) but I didn't find one triggered on a new rollout start. That trigger seems useful for [Grafana annotations](https://argo-rollouts.readthedocs.io/en/release-1.5/generated/notification-services/grafana/) if we want to catch a period between rollout start and completion/abort. Any ideas?

**Comment by danil-smirnov** [comment, marked as answer]
I've created a custom trigger like this:
```
      trigger.on-rollout-started: |
        - send: [rollout-started]
          when: rollout.status.phase == 'Progressing' and rollout.status.currentStepIndex == 0
```
Maybe it can be included in the list of default triggers?

**Reply by zachaller** [reply]
Just to add another option as well
```
trigger.on-rollout-updated-webhook: |
    - send: [rollout-json]
      when: rollout.status != nil && rollout.status.stableRS != rollout.status.currentPodHash
      oncePer: rollout.metadata.annotations["rollout.argoproj.io/revision"]
```

**Reply by zachaller** [reply]
There is also the event based version: https://github.com/argoproj/argo-rollouts/blob/master/manifests/notifications/on-rollout-updated.yaml

**Reply by danil-smirnov** [reply]
Thank you @zachaller but as I see in the [docs](https://argo-rollouts.readthedocs.io/en/stable/features/notifications/#default-trigger-templates):

`on-rollout-updated when a rollout definition is changed`

I'm not sure how it's related to the rollout started event...
